client/src/Pages/Profile/UserInfo/index.js

```diff
@@ -0,0 +1,124 @@
+import React, { useState, useEffect } from 'react';
+import PropTypes from 'prop-types';
+import axios from 'axios';
+import {
+  Typography,
+  Avatar,
+  Grid,
+  Paper,
+  CircularProgress,
+} from '@material-ui/core';
+import { Alert } from '@material-ui/lab';
+
+import Button from '../../../Components/Button';
+
+import useStyles from './style';
+
+function UserInfo({ getUserName }) {
+  const classes = useStyles();
+  const [user, setUser] = useState({});
+  const [loading, setLoading] = useState('');
+  const [errorMsg, setErrorMsg] = useState('');
+
+  const fetchingData = async (isCurrent) => {
+    try {
+      setLoading(true);
+      const { data } = await axios.get(`api/v1/users`);
+      if (isCurrent) {
+        setUser(data.data);
+        getUserName(data.data.username);
+        setLoading(false);
+      }
+    } catch (error) {
+      setLoading(false);
+      setErrorMsg(error.response.data);
+    }
+  };
+
+  useEffect(() => {
+    let isCurrent = true;
+    fetchingData(isCurrent);
+    return () => {
+      isCurrent = false;
+    };
+  }, []);
+
+  return (
+    <div className={classes.root}>
+      <Typography variant="h2">User Information</Typography>
+      {loading ? (
+        <CircularProgress />
+      ) : (
+        <>
+          <Grid lg="12" justify="center">
+            <Avatar className={classes.avatar}>
+              {user.username && user.username.slice(0, 1).toUpperCase()}
+            </Avatar>
```

Comment:
l like the way you handle loading and error :+1:

Reply:
Yes, I totally agree!